sensor/kubernetes/eventpipeline/component/message.go

```diff
@@ -0,0 +1,52 @@
+package component
+
+import (
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+)
+
+// CompatibilityDetectionMessage should be used by old handlers
+// it's here for retrocompatibility reasons.
+type CompatibilityDetectionMessage struct {
+	// Object The deployment object that needs to be processed by the detector
+	Object *storage.Deployment
+	// Action the event action (CREATE_RESOURCE, REMOVE_RESOURCE, UPDATE_RESOURCE, and SYNC_RESOURCE)
+	Action central.ResourceAction
+}
+
+// ResourceEvent message used by the event pipeline's components
+type ResourceEvent struct {
+	// ForwardMessages messages generated by the handlers that need to be forwarded to central
+	ForwardMessages []*central.SensorEvent
+	// CompatibilityDetectionDeployment should be used by old handlers
+	// and it's here for retrocompatibility reasons.
+	// This property should be removed in the future and only the
+	// DetectionObject should be sent
```

Comment:
I think this comment was written based on the first iteration of this. 
```suggestion
	// deployment references should be sent
```